src/renderer/pages/sshsetup/GenerateKey.js

```diff
@@ -44,42 +61,59 @@ const GenerateKey = ({ onNext }) => {
       data: generateKeySuccess,
     },
     dispatch,
-  ] = useReducer(fetchReducer, {
+  ] = React.useReducer(fetchReducer, {
     isLoading: false,
     isError: false,
     data: null,
   });
 
   // Effect to save user details like username , email and bitbucket_uuid in our auth store.
-  useEffect(() => {
-    setAuthState(prevState => ({
-      ...prevState,
-      username,
-      email,
-      bitbucket_uuid,
-    }));
-  }, [username, email, bitbucket_uuid]);
+  React.useEffect(() => {
+    if (username) sessionStore.addUserName(username);
+    if (email) sessionStore.addEmail(email);
+    if (userProvidedEmail) sessionStore.addEmail(userProvidedEmail);
+    if (bitbucket_uuid) sessionStore.addBitbucketUUID(bitbucket_uuid);
+    if (avatar_url) sessionStore.addAvatarUrl(avatar_url);
+  }, [username, email, bitbucket_uuid, userProvidedEmail, avatar_url]);
 
   // Effect run to check if key for "selectedProvider" and "username" exists in ssh/config folder
-  useEffect(() => {
-    async function checkIfKeyAlreadyExists(selectedProvider, username) {
-      const keyAlreadyExists = await window.ipc.callMain(
+  React.useEffect(() => {
+    async function checkIfKeyAlreadyExists(selectedProvider, mode, username) {
+      const keyAlreadyExistsInFileSystem = await window.ipc.callMain(
         'check-key-already-exists',
         {
           selectedProvider,
+          mode,
           username,
         }
       );
-      setKeyAlreadyExists(keyAlreadyExists);
+      setKeyAlreadyExistsInFileSystem(keyAlreadyExistsInFileSystem);
     }
 
-    if (selectedProvider && username) {
-      checkIfKeyAlreadyExists(selectedProvider, username);
+    if (sessionStore.provider && username) {
```

Comment:
We should keep this logic until we add regenerate key step from Home screen. Once we add that we can stop user from generating key based on whether selected provider and username combination exists in store or not.

Reply:
Sure.